pkg/controller/deployment/deployment_controller.go

```diff
@@ -1036,6 +1036,7 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 	newDeployment := deployment
 	// TODO: Reconcile this with API definition. API definition talks about ready pods, while this just computes created pods.
 	newDeployment.Status = extensions.DeploymentStatus{
+		ObservedGeneration:  int(deployment.Generation),
```

Comment:
We need to ensure that if this is retried (doesn't look like we do that now), we don't pick up a new Generation value. At least put a comment.